Golem/ActivityLoop.cs

```diff
@@ -67,53 +65,46 @@ public ActivityLoop(HttpClient httpClient, CancellationToken token, ILogger logg
                     await foreach (string json in EnumerateMessages(reader).WithCancellation(_token))
                     {
                         _logger.LogInformation("got json {0}", json);
-                        var activity_state = parseActivityState(json);
-                        if (activity_state?.AgreementId == null)
+                        var activityStates = parseActivityStates(json);
+
+                        List<Job> currentJobs = await updateJobs(activityStates, jobs);
+
+                        if (currentJobs.Count == 0)
                         {
-                            _logger.LogDebug("No activity");
-                            setAllJobsFinished();
+                            _logger.LogDebug("Cleaning current job field");
                             setCurrentJob(null);
-                            continue;
+                            //TODO why jobs terminated as Idle stay Idle?
+                            jobs.SetAllJobsFinished();
```

Comment:
I broke something while encapsulating functions. It will get fixed while working on getting statuses from API.